src/Overlay.tsx

```diff
@@ -0,0 +1,123 @@
+import styled from 'styled-components'
+import React, {ReactElement} from 'react'
+import {get, COMMON, POSITION, SystemPositionProps, SystemCommonProps} from './constants'
+import {ComponentProps} from './utils/types'
+import {useOverlay, AnchoredPositionHookSettings, TouchOrMouseEvent} from './hooks'
+import Portal from './Portal'
+import sx, {SxProp} from './sx'
+
+type StyledOverlayProps = {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+  visibility: 'visible' | 'hidden'
+}
+
+const heightMap = {
+  sm: '480px',
+  md: '640px',
+  auto: 'auto'
+}
+
+const widthMap = {
+  sm: '256px',
+  md: '320px',
+  lg: '480px',
+  xl: '640px',
+  auto: 'auto'
+}
+
+/*TODO replace with shadow functional color variable when it's shipped to primer/primitives*/
+const StyledOverlay = styled.div<StyledOverlayProps & SystemCommonProps & SystemPositionProps & SxProp>`
+  background-color: ${get('colors.bg.overlay')};
+  box-shadow: ${get('overlay.boxShadow')};
+  position: absolute;
+  min-width: 192px;
+  max-width: 640px;
+  height: ${props => heightMap[props.height || 'auto']};
+  width: ${props => widthMap[props.width || 'auto']};
+  border-radius: ${get('overlay.borderRadius')};
+  overflow: hidden;
+  animation: overlay-appear 200ms ${get('animation.easeOutCubic')};
+
+  @keyframes overlay-appear {
+    0% {
+      opacity: 0;
+      transform: translateY(${get('space.2')});
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+  visibility: ${props => props.visibility};
+  ${COMMON};
+  ${POSITION};
+  ${sx};
+`
+export type OverlayProps = {
+  ignoreClickRefs: React.RefObject<HTMLElement>[]
```

Comment:
This (`ignoreClickRefs`) should be optional, to align with [`useOverlay`](https://github.com/primer/components/pull/1084/files#r604260434) and `useOnOutsideClick`: https://github.com/primer/components/blob/46c6b804ea38d29f58a4c1a92238a6a6c3729667/src/hooks/useOnOutsideClick.tsx#L7